subsys/bluetooth/controller/hci/hci.c

```diff
@@ -7087,6 +7087,15 @@ static void le_per_adv_sync_report(struct pdu_data *pdu_data,
 				 * mark as partial data.
 				 */
 				data_status = BT_HCI_LE_ADV_EVT_TYPE_DATA_STATUS_PARTIAL;
+			} else if (ftr->aux_sched &&
+				   (data_len_total == CONFIG_BT_CTLR_SCAN_DATA_LEN_MAX) &&
+				   (cte_type != BT_HCI_LE_NO_CTE)) {
+				/* No data left but: have AuxPtr, scheduled aux scan and CTE is
+				 * included in a PDU, there is another chained PDU that is empty
+				 * but has CTE.
+				 */
+				data_status = BT_HCI_LE_ADV_EVT_TYPE_DATA_STATUS_COMPLETE;
```

Comment:
If there's another PDU to come, isn't it supposed to be a `BT_HCI_LE_ADV_EVT_TYPE_DATA_STATUS_PARTIAL` here as well?

Reply:
Good question... If `CONFIG_BT_CTLR_SCAN_DATA_LEN_MAX` is set to a lower value than 1650 bytes, then the `data_status` can be wrong also. 
It look like the `data_status` can be correctly set only when we have received next chained PDU.
Hence reports should be generated for last but one PDU.